diff --git a/CMakeLists.txt b/CMakeLists.txt
index 007c0a2..c38eaaf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -91,8 +91,12 @@ endif (TLSUV_HTTP)
 
 add_subdirectory(src/${TLSUV_TLSLIB})
 
-find_package(libuv CONFIG QUIET)
-if (libuv_FOUND)
+if (TARGET libuv::uv)
+    message(NOTICE "upstream project set libuv target")
+    set(TLSUV_LIBUV_LIB libuv::uv)
+    set(libuv_FOUND TRUE)
+else()
+    find_package(libuv CONFIG QUIET)
     # newer libuv versions (via VCPKG) have proper namespacing
     if (TARGET libuv::uv_a)
         set(TLSUV_LIBUV_LIB libuv::uv_a)
@@ -103,7 +107,9 @@ if (libuv_FOUND)
     else()
         set(TLSUV_LIBUV_LIB uv)
     endif()
-else()
+endif ()
+
+if (NOT libuv_FOUND)
     pkg_check_modules(libuv REQUIRED IMPORTED_TARGET libuv)
     set(TLSUV_LIBUV_LIB PkgConfig::libuv)
 endif()
diff --git a/src/openssl/CMakeLists.txt b/src/openssl/CMakeLists.txt
index b485374..dab95aa 100644
--- a/src/openssl/CMakeLists.txt
+++ b/src/openssl/CMakeLists.txt
@@ -1,5 +1,3 @@
-find_package(OpenSSL REQUIRED)
-
 add_library(openssl-impl OBJECT
         engine.c
         keys.c
@@ -8,22 +6,27 @@ add_library(openssl-impl OBJECT
 target_include_directories(openssl-impl
         PRIVATE ${PROJECT_SOURCE_DIR}/include)
 if (WIN32)
-target_compile_definitions(openssl-impl
-        PRIVATE WINVER=0x0A00
-        PRIVATE _WIN32_WINNT=0x0A00
-        PRIVATE _CRT_SECURE_NO_WARNINGS
-        PRIVATE _CRT_NONSTDC_NO_DEPRECATE
-        PRIVATE _WINSOCK_DEPRECATED_NO_WARNINGS
-        PRIVATE WIN32_LEAN_AND_MEAN
-)
-if (MSVC)
-    target_compile_options(openssl-impl
-            PRIVATE /wd4100 # disable "unreferenced formal parameter" warning
+    target_compile_definitions(openssl-impl
+            PRIVATE WINVER=0x0A00
+            PRIVATE _WIN32_WINNT=0x0A00
+            PRIVATE _CRT_SECURE_NO_WARNINGS
+            PRIVATE _CRT_NONSTDC_NO_DEPRECATE
+            PRIVATE _WINSOCK_DEPRECATED_NO_WARNINGS
+            PRIVATE WIN32_LEAN_AND_MEAN
     )
-endif (MSVC)
+    if (MSVC)
+        target_compile_options(openssl-impl
+                PRIVATE /wd4100 # disable "unreferenced formal parameter" warning
+        )
+    endif (MSVC)
+endif ()
 
+if (NOT TARGET OpenSSL::SSL)
+    find_package(OpenSSL)
 endif ()
 
-target_link_libraries(openssl-impl PRIVATE OpenSSL::SSL)
+if (TARGET OpenSSL::SSL)
+    target_link_libraries(openssl-impl PRIVATE OpenSSL::SSL)
+endif ()
 
 
